### PR TITLE
feat: Add requestPinWidget method

### LINKED ISSUE #191: Pinning widget
It might be nice to add this feature https://developer.android.com/develop/ui/views/appwidgets/configuration#pin

### PR DESCRIPTION
**Pull request by tenhobi**
Fixes #191

### CODE REVIEW

**Comment by tenhobi** [comment]
It would be awesome to add a Dart callback as a parameter so we can handle successfully pinning a widget. But I am not sure how to do it, could you help @ABausG? An example of the intent is in https://developer.android.com/develop/ui/views/appwidgets/configuration#pin

File: lib/home_widget.dart [hunk 1]
```diff
@@ -50,6 +50,27 @@ class HomeWidget {
     });
   }
 
+  /// Pins the HomeScreen Widget
```
Comment (petrnymsa) [inline lib/home_widget.dart:53]: Add docs when result `true` is returned, e.g only when widget is successfully pinned

File: lib/home_widget.dart [hunk 2]
```diff
@@ -50,6 +50,27 @@ class HomeWidget {
...
+  ///
+  /// Android Widgets will look for [qualifiedAndroidName] then [androidName] and then for [name]
+  /// There is no iOS alternative
+  ///
+  /// [qualifiedAndroidName] will use the name as is to find the WidgetProvider
+  /// [androidName] must match the classname of the WidgetProvider, prefixed by the package name
+  static Future<bool?> pinWidget({
```
Comment (petrnymsa) [inline lib/home_widget.dart:60]: rename to `requestPinWidget` - we are requesting system to pin it, we don't have control over it (low android version, launcher does not support it.

**Comment by petrnymsa** [comment]
LGTM! Tested in our application. Can you look at it @ABausG ? Thanks 🙏

**Comment by tenhobi** [comment]
Not really sure what the last iOS integration issue is, it seems unrelated to the PR.
image

File: README.md [hunk 3]
```diff
@@ -446,6 +446,23 @@ WorkmanagerPlugin.setPluginRegistrantCallback { registry in
 ```
 to [AppDelegate.swift](example/ios/Runner/AppDelegate.swift)
 
+### Request Pin Widget
+Requests pin of HomeWidget to home screen directly from the app.
```
Comment (ABausG) [inline README.md:450]: Maybe add a quick sentence that that would add the Widget to the Homescreen. (I had to read that Android Docs blurb twice to understand that `pinning == adding`)
```suggestion
Requests to add the Widget to the users HomeScreen by pinning it to the Users HomeScreen.
```

File: lib/home_widget.dart [hunk 4]
```diff
@@ -50,6 +50,32 @@ class HomeWidget {
     });
   }
 
+  /// Determines whether pinning HomeScreen Widget is supported.
+  static Future<bool?> isRequestPinWidgetSupported() {
+    return _channel.invokeMethod('isRequestPinWidgetSupported');
+  }
+
+  /// Pins the HomeScreen Widget.
```
Comment (ABausG) [inline lib/home_widget.dart:58]:
```suggestion
  /// Requests to Pin (Add) the HomeScreenWidget to the User's Home Screen
  ///
  /// This is supported only on some Android Launchers and only with Android API 26+
```

File: test/home_widget_test.dart [hunk 5]
```diff
@@ -47,6 +47,10 @@ void main() {
           return Future.value(launchUri);
         case 'registerBackgroundCallback':
           return true;
+        case 'requestPinWidget':
+          return true;
```
Comment (ABausG) [inline test/home_widget_test.dart:51]: Should maybe return `null` in order to mimic better what the actual result is doing.

**Review by ABausG — COMMENTED** [review]
Thanks for this! Just some minor doumentation nibs.

Integration Failure is indeed not your fault. The Workflow did test the iOS part on flutter beta and it seems `ViewConfiguration` will change to named arguments.

I remember you talking about potentially adding a dart callback to the intent. While this possibly might be useful I don't think it is necessary to support this from the beginning and thus should not stop this from being added.

Could you also target the `dev` branch for this?

**Comment by tenhobi** [comment]
Awesome, will do tomorrow 💪

File: README.md (hunk 3 above)
Comment (petrnymsa) [reply README.md:450]: I think we should left "pin" word here as well, to match Android's native world, something like 

````
Requests to add (Android's pin widget feature) the Widget to the users HomeScreen by pinning it to the Users HomeScreen. 
```
or something like that

Comment (ABausG) [reply README.md:450]: Maybe let's go with the wording I proposed in the other file to have `Pin (Add)`.

Good point of keeping the native wording. Thanks @petrnymsa

**Comment by tenhobi** [comment]
Hi @ABausG, I just finished all the suggestions :)

**Review by ABausG — APPROVED** [review]

**Comment by tenhobi** [comment]
So, can this be merged? 😊🙏

**Pull request merged**